Multiple user toolchains (#2340)
conan-io/docs: reference/conanfile/tools/cmake/cmaketoolchain.rst

There are two ways of providing a custom CMake toolchain file:

- The ``conan_toolchain.cmake`` file can be completely skipped and replaced by a user one, defining the

>   ``tools.cmake.cmaketoolchain:toolchain_file=<filepath>`` configuration value

Reviewer: Nop, the correct syntax is "a user", not "an user"

- A custom user toolchain file can be added (included from) the ``conan_toolchain.cmake`` one, by using the
  ``user_toolchain`` block described below, and defining the ``tools.cmake.cmaketoolchain:user_toolchain=<filepath>``
  configuration value.

  The configuration ``tools.cmake.cmaketoolchain:user_toolchain=<filepath>`` can be defined in the :ref:`global.conf<global_conf>`
  but also creating a Conan package for your toolchain and using ``self.conf_info`` to declare the toolchain file:

    .. code:: python

        import os
        from conans import ConanFile
        class MyToolchainPackage(ConanFile):
            ...
            def package_info(self):
                f = os.path.join(self.package_folder, "mytoolchain.cmake")
                self.conf_info["tools.cmake.cmaketoolchain:user_toolchain"] = f

>   If you declare a `tool_require` to the package declared above, the toolchain will be automatically applied.

Reviewer 2: ```suggestion
  If you declare the previous package as a ``tool_require``, the toolchain will be automatically applied.
```

> - You can also apply several user toolchains. If you have more than one tool requires, you can gather the values

Reviewer: ```suggestion
- You can also apply several user toolchains. If you have more than one ``tool_requires``, you can gather the values
```
Otherwise it reads a bit weird
